internal/cloud/realm/user.go

```diff
@@ -92,11 +93,139 @@ type User struct {
 // UserIdentity is a Realm application user identity
 type UserIdentity struct {
 	UID          string                 `json:"id"`
-	ProviderType string                 `json:"provider_type"`
+	ProviderType ProviderType           `json:"provider_type"`
 	ProviderID   primitive.ObjectID     `json:"provider_id"`
 	ProviderData map[string]interface{} `json:"provider_data,omitempty"`
 }
 
+// ProviderType is a Realm application provider type
+type ProviderType string
+
+// set of supported provider type values
+const (
+	ProviderTypeEmpty          ProviderType = ""
+	ProviderTypeUserPassord    ProviderType = "local-userpass"
+	ProviderTypeAPIKey         ProviderType = "api-key"
+	ProviderTypeFacebook       ProviderType = "oauth2-facebook"
+	ProviderTypeGoogle         ProviderType = "oauth2-google"
+	ProviderTypeAnonymous      ProviderType = "anon-user"
+	ProviderTypeCustomToken    ProviderType = "custom-token"
+	ProviderTypeApple          ProviderType = "oauth2-apple"
+	ProviderTypeCustomFunction ProviderType = "custom-function"
+)
+
+// slice of supported provider type values as string slice
```

Comment:
either reword this to be more general: `// set of supported auth constants`, or move the comment down to be right above `ValidProviderTypes`, and make it something like `// ValidProviderTypes is the set of supported...`

I lean towards the former approach, but your call here